tinygrad/lazy.py

```diff
@@ -210,6 +210,13 @@ def shuffle_and_prune_movement_ops(self, st: ShapeTracker, op: MovementOps, arg:
   def reduce_op(self:LazyBuffer, op:ReduceOps, new_shape:Tuple[int, ...]) -> LazyBuffer:
     if self.shape == tuple(new_shape): return self
     srcs = _push_movement_ops((self,)) if SHUFFLE_MOVEMENT_OPS else (self,)
+        
+    differences = [a != b for a,b in zip(self.shape, new_shape)]
+    if sum(differences) > 1:
+      last_difference = len(differences) - 1 - differences[::-1].index(True)
+      intermediate_shape = tuple(new if i == last_difference else old for i, (new, old) in enumerate(zip(new_shape, self.shape)))
+      return create_lazybuffer(self.device, ShapeTracker(intermediate_shape), ReduceOps, LazyOp(op, srcs, intermediate_shape), self.dtype).reduce_op(op, new_shape)
```

Comment:
Finds instances where shape has differences on multiple axis, and 'clamps' the reduction to only reduce along the final axis that needs reducing. Recursively calls `.reduce_op()` handle this.

Presuming we'll need to **not** do this on other accelerators, and intuitively this would be 'nicer' around linearizer.py, but the linearizer et al do not expect multiple reductions.